packages/botfuel-dialog/src/dialog-manager.js

```diff
@@ -28,11 +28,16 @@ const DialogError = require('./errors/dialog-error');
 class DialogManager extends Resolver {
   /**
    * @constructor
+   * @param {Object} bot - the bot
    * @param {Object} brain - the bot brain
    * @param {Object} config - the bot config
    */
-  constructor(brain, config) {
+  constructor(bot) {
+    const { brain, config } = bot;
+
     super(config, 'dialog');
+
+    this.bot = bot;
```

Comment:
is it really needed to have both?

Reply:
Yes because:
* we need the `bot` to build the dialogs
* we need the brain to manipulate dialogs data